src/infrastructure/content/NetworkMessageHandler.java

```diff
@@ -0,0 +1,247 @@
+	 * Networking module will be calling this method with json string message to send data to the content module
+	 *
+	 * creating error log message if message in not converting into json object,
+	 * or if not able extract field from object, or if not able to extract handler from hashmap (handlerMap)
+	 * if message will be successfully sent to UI, then creating success log message
+	 */
+	@override
+	public void onMessageReceived(String message) {
+		logMessage = "onMessageReceived method of NetworkMessageHandler class is executing";
+		logger.log(ModuleID.INFRASTRUCTURE, LogLevel.INFO, logMessage);
+		
+		try {
+			JSONObject jsonObject = new JSONObject(message);
+		} 
+		catch(Exception e) {
+			logMessage = "converting JSON string to JSON object process failed";
+			logger.log(ModuleID.INFRASTRUCTURE, LogLevel.ERROR, logMessage);
+			return;
+		}
+		
+		try {
+			metafield = jsonObject.getString("meta");
+		}
+		catch(Exception e) {
+			logMessage = "failed to get metafield value of argument message";
+			logger.log(ModuleID.INFRASTRUCTURE, LogLevel.ERROR, logMessage);
+			return;
+		}
+		
+		try {
+			handler = handlerMap.get("UI");
+		}
+		catch(Exception e) {
+			logMessage = "failed to get IContentNotification handler from handlerMap";
+			logger.log(ModuleID.INFRASTRUCTURE, LogLevel.ERROR, logMessage);
+			return;
+		}
+		
+		jsonObject.remove("meta");
+		
+		if(metafield.equals("newUser")) {
+			logMessage = "New User case";
+			logger.log(ModuleID.INFRASTRUCTURE, LogLevel.INFO, logMessage);
+			
+			imagemap = new HashMap<String, String>();
+			jsonObjectArray = new JSONArray();
+			
+			try {
+				jsonObjectArray = jsonObject.getJSONArray("imageMap");
+			}
+			catch(Exception e) {
+				logMessage = "failed to get imageMap field value of argument message";
+				logger.log(ModuleID.INFRASTRUCTURE, LogLevel.ERROR, logMessage);
+				return;
+			}
+			
+			jsonObject.remove("imageMap");
+			for(int i = 0; i < jsonObjectArray.size(); i++) {
+				try {
+					object = jsonObjectArray.getJSONArray(i);
+				}
+				catch(Exception e) {
+					logMessage = "failed to get element of json array";
+					logger.log(ModuleID.INFRASTRUCTURE, LogLevel.ERROR, logMessage);
+					return;
+				}
+				
+				try {
+					username = object.getString("username");
+				}
+				catch(Exception e) {
+					logMessage = "failed to get username field value from object";
+					logger.log(ModuleID.INFRASTRUCTURE, LogLevel.ERROR, logMessage);
+					return;
+				}
+				
+				try {
+					userimage = object.getString(username);
+				}
+				catch(Exception e) {
+					logMessage = "failed to get image from imageMap";
+					logger.log(ModuleID.INFRASTRUCTURE, LogLevel.ERROR, logMessage);
+					return;
+				}
+				imagemap.put(username, userimage);
+				if(!imageMap.containsKey(username)) {
+					jsonObject.put("username", username);
+					jsonObject.put("image", userimage);
+				}
+			}
+			
+			ContentCommunicator.setImageMap(imagemap);
```

Comment:
Required type: `java.util.HashMap`
Provided type : `HashMap`

Reply:
I believe this was resolved